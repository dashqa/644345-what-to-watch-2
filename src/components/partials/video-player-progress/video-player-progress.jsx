import React from "react";
import PropTypes from "prop-types";

import withPlayerDuration from "@hocs/with-player-duration/with-player-duration";

const VideoPlayerProgress = ({progressPosition, progressRef, toggleRef, timing}) => {
  const toggleStyles = {
    left: `${progressPosition}%`
  };

  return (
    <div className="player__controls-row">
      <div className="player__time">
        <progress
          ref={progressRef}
          value={progressPosition}
          className="player__progress"
          max="100">
        </progress>
        <div
          ref={toggleRef}
          className="player__toggler"
          style={toggleStyles}
        >
          Toggler</div>
      </div>
      <div className="player__time-value">{timing}</div>
    </div>
  );
};

VideoPlayerProgress.defaultProps = {
  progressPosition: 0,
  progressRef: {},
  toggleRef: {},
  timing: ``,
};

VideoPlayerProgress.propTypes = {
  progressPosition: PropTypes.number.isRequired,
  progressRef: PropTypes.object.isRequired,
  toggleRef: PropTypes.object.isRequired,
  timing: PropTypes.string.isRequired,
};

export {VideoPlayerProgress};

export default withPlayerDuration(VideoPlayerProgress);
