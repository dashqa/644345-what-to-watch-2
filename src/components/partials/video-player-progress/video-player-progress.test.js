import React from 'react';
import renderer from 'react-test-renderer';
import {VideoPlayerProgress} from './video-player-progress';

const props = {
  progressPosition: 0,
  progressRef: React.createRef(),
  toggleRef: React.createRef(),
  timing: `00:00:00`,
};

it(`Video player progress correctly renders after relaunch`, () => {
  const tree = renderer.create(<VideoPlayerProgress {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
