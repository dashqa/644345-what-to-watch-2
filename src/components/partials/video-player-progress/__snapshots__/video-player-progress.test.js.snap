// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video player progress correctly renders after relaunch 1`] = `
<div
  className="player__controls-row"
>
  <div
    className="player__time"
  >
    <progress
      className="player__progress"
      max="100"
      value={0}
    />
    <div
      className="player__toggler"
      style={
        Object {
          "left": "0%",
        }
      }
    >
      Toggler
    </div>
  </div>
  <div
    className="player__time-value"
  >
    00:00:00
  </div>
</div>
`;
